### ISSUE TITLE
feat: use grain for data loading

### ISSUE THREAD

**Issue by fbanelli** [antonioterpin/synthpix#206]
**Is your feature request related to a problem? Please describe.**
The prefetching queue often leads to problems, let's use some standard data-loader. [Grain](https://github.com/google/grain) by google is a very dependency-light option which fits well within the jax ecosystem.

**Describe the solution you'd like**
The whole scheduler part of the code can be easily replaced by a grain DataLoader.

**Describe alternatives you've considered**
Alternatives are PyTorch DataLoaders but require pytorch as a dependency, or TensorFlow data sources, but along with the rest of TensorFlow they aren't as supported anymore.
